### PR TITLE
Add ingestion mutation for SLSA

### PR DESCRIPTION
**Pull request by mihaimaruseac**
Need to make sure we use `XXXSpec` for read-only queries and `XXXInput` for mutation, since they have different nullability rules and semantics. Took a while to found bugs caused by using the wrong one (after merge conflict resolution).

While here also fix a typo in a file name and remove leftover debug print

Note: `pkg/assembler/backends/testing/hasSLSA.go` has been rewritten a lot, so github might hide it. That actually contains most of this PR

### CODE REVIEW

File: pkg/assembler/backends/neo4j/certifyVuln.go [hunk 1]
```diff
@@ -401,7 +400,6 @@ func (c *neo4jClient) IngestVulnerability(ctx context.Context, pkg model.PkgInpu
 		sb.WriteString(merge)
 		sb.WriteString(returnValue)
 
-		fmt.Println(sb.String())
```
Comment (pxp928) [inline pkg/assembler/backends/neo4j/certifyVuln.go:404]: you can remove this from your PR if you want. it's part of PR #552

File: pkg/assembler/backends/testing/backend.go [hunk 2]
```diff
@@ -114,10 +114,6 @@ func GetBackend(args backends.BackendArgs) (backends.Backend, error) {
 	if err != nil {
 		return nil, err
 	}
-	err = registerAllHasSLSA(client)
```
Comment (pxp928) [inline pkg/assembler/backends/testing/backend.go:117]: do we no longer want to add data to the testing backend?

**Review by pxp928 — APPROVED** [review]
LGTM

File: pkg/assembler/backends/testing/backend.go (hunk 2 above)
Comment (pxp928) [reply pkg/assembler/backends/testing/backend.go:117]: Does this need to be replaced by another? Or ingestion via the client only?

Comment (mihaimaruseac) [reply pkg/assembler/backends/testing/backend.go:117]: Via client/playground is ok here.

Don't remove the others yet as we might use them for testing/perf

File: pkg/assembler/backends/neo4j/certifyVuln.go (hunk 1 above)
Comment (mihaimaruseac) [reply pkg/assembler/backends/neo4j/certifyVuln.go]: Dropped that specific commit from this branch

**Review by lumjjb — APPROVED** [review]

**Pull request merged**